src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -628,32 +621,48 @@ namespace winrt::TerminalApp::implementation
     // Method Description:
     // - Dispatch the current search text as a ExecuteCommandline action.
     // Arguments:
-    // - <none>
+    // - filteredCommand - Selected filtered command - might be null
     // Return Value:
     // - <none>
-    void CommandPalette::_dispatchCommandline()
+    void CommandPalette::_dispatchCommandline(winrt::TerminalApp::FilteredCommand const& command)
     {
-        auto cmdline{ _getTrimmedInput() };
-        if (cmdline.empty())
+        const auto filteredCommand = command ? command : _buildCommandLineCommand(_getTrimmedInput());
+        if (filteredCommand.has_value())
         {
-            return;
-        }
+            if (_commandLineHistory.Size() == CommandLineHistoryLength)
+            {
+                _commandLineHistory.RemoveAtEnd();
+            }
+            _commandLineHistory.InsertAt(0, filteredCommand.value());
```

Comment:
@zadjii-msft - I took this approach because I didn't know what is the best behavior here, and in such cases I choose the simplest one. The options I considered were:
* Set, sorted alphabetically
* Set, sorted by MRU
* List, sorted by MRU
* A combination that you mentioned of don't push the command if it equals to the previous one

If you have a strong feeling about one of those, I can change this now or in followup (or we can add configs and leave to the user).